src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/Authorization/OrganizationUserUserDetailsAuthorizationHandler.cs

```diff
@@ -49,4 +53,18 @@ private async Task<bool> CanReadAllAsync(Guid organizationId)
         // Allow provider users to read all organization users if they are a provider for the target organization
         return await _currentContext.ProviderUserForOrgAsync(organizationId);
     }
+
+    private bool ReadWithAccountRecovery(Guid organizationId)
+    {
+        var organization = _currentContext.GetOrganization(organizationId);
+        if (organization is
+        { Permissions.ManageUsers: false } and
+        { Permissions.ManageResetPassword: true }
```

Comment:
Not strictly true I think. If you have `ManageUsers && ManageResetPassword`, you have _permission_ to access this data, even if in practice you're going to receive the full data set. But that's for the controller to manage - the authorization code itself shouldn't have any knowledge of that.

This should allow:
- owners
- admins
- ManageResetPassword custom permission
- providers